components/base/sensorcontrolled/velocities.go

```diff
@@ -0,0 +1,189 @@
+package sensorcontrolled
+
+import (
+	"context"
+	"time"
+
+	"github.com/golang/geo/r3"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/control"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+// TODO: RSDK-5355 useControlLoop bool should be removed after testing.
+const useControlLoop = false
+
+func setupControlLoops(sb *sensorBase) error {
+	// TODO: RSDK-5355 useControlLoop bool should be removed after testing
+	if useControlLoop {
+		loop, err := control.NewLoop(sb.logger, controlLoopConfig, sb)
+		if err != nil {
+			return err
+		}
+		sb.loop = loop
+
+		if sb.loop != nil {
+			if err := sb.loop.Start(); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (sb *sensorBase) SetVelocity(
+	ctx context.Context, linear, angular r3.Vector, extra map[string]interface{},
+) error {
+	sb.opMgr.CancelRunning(ctx)
+	// check if a sensor context has been started
+	if sb.sensorLoopDone != nil {
+		sb.sensorLoopDone()
+	}
+
+	// set the spin loop to false, so we do not skip the call to SetState in the control loop
+	sb.setPolling(false)
+
+	// start a sensor context for the sensor loop based on the longstanding base
+	// creator context, and add a timeout for the context
+	timeOut := 10 * time.Second
+	var sensorCtx context.Context
+	sensorCtx, sb.sensorLoopDone = context.WithTimeout(context.Background(), timeOut)
+
+	// TODO: RSDK-5355 remove control loop bool after testing
+	if useControlLoop && sb.loop != nil {
+		// if we have a loop, let's use the SetState function to call the SetVelocity command
+		// through the control loop
+		sb.logger.Info("using loop")
+		sb.pollsensors(sensorCtx, extra)
+		return nil
+	}
+
+	sb.logger.Info("setting velocity without loop")
+	// else do not use the control loop and pass through the SetVelocity command
+	return sb.controlledBase.SetVelocity(ctx, linear, angular, extra)
+}
+
+func (sb *sensorBase) pollsensors(ctx context.Context, extra map[string]interface{}) {
```

Comment:
This function can go away now or later - it is a helper that polls the linear and angular velocity sensors from within the sensor-controlled base. I'm thinking of just having it run based on the use control loop and removing the check for `polling`. The `isPolling` functionality is related to stopping the `Spin` control loop (not using the control package) from activity when other API calls are made.